smart_campus/app/models.py

```diff
@@ -33,6 +33,7 @@ def create_user(self, email, password=None, nickname=""):
         )
 
         user.set_password(password)
+        user.full_clean()
         user.save(using=self._db)
```

Comment:
Could you please briefly describe what `uesr.full_clean()` and `(using=self._db)` is for me?
Thanks in advance!

Reply:
full_clean() will check if the data is valid for the fields in the model while save() will not, so it's possible for invalid data to enter the DB.

This issue if mainly for the EmailField in the User model.
Without full_clean(), invalid email will be directly saved.

Reply:
Found that self._db is None, it can be removed

Reply:
I see. Thanks!

Reply:
The ValidationError will not only raise cause email error input, but also will raise with max_length like nickname input.
I think we can just print the ValidationError's message, you can check out with email validator or django source

Reply:
Okay